### PR TITLE
Refactor world select

### LINKED ISSUE #282: Set default background when room isn't loaded
Use radial gradient from the home page

### LINKED ISSUE #283: Use room alias instead of unnamed room when avaliable

### LINKED ISSUE #291: broken homeworld selection in world list when leaving a room via `alt+L`
1. leave any room via alt+L
2. notice homeworld is always selected after leaving any room
3. click Load World, notice nothing happens
4. select any other room, Load World now works again

### LINKED ISSUE #319: Home World selected / onboarding modal visible when not in home world

### LINKED ISSUE #350: Joining a room takes too many button presses
Having to press 'load world' and then 'enter world' feels completely redundant. Shouldn't you automatically enter as soon as the world has loaded?

### PR DESCRIPTION
**Pull request by ajbura**
TODO:
- [x] Refactor world select
- [x] Fix world select UX
- [x] Fix portal UX

Fixes #350 
Fixes #319 
Fixes #291 
Fixes #283 
Fixes #282

### CODE REVIEW

**Review by robertlong — CHANGES_REQUESTED** [review]

**Comment by robertlong** [comment]
- Create World Dialog should wait for world to be created and then redirect so we don't have that weird state for `RoomBeingCreated`
- We should show the scene name and creator above the loading world progress bar

File: src/ui/App.tsx [hunk 1]
```diff
@@ -53,7 +53,7 @@ const LoginView = lazy(() => import("./views/login/LoginView"));
 const GLTFViewer = lazy(() => import("./views/gltf-viewer/GLTFViewer"));
 const AssetPipeline = lazy(() => import("./views/asset-pipeline/AssetPipeline"));
 const SessionView = lazy(() => import("./views/session/SessionView"));
-const WorldView = lazy(() => import("./views/session/world/WorldView"));
+const World = lazy(() => import("./views/session/world/World"));
```
[the remark this answers is not in the document]
Comment (ajbura) [reply src/ui/App.tsx]: Changed this to WorldRootView as we already have a WordView component (Used inside WorldRootView).

File: src/ui/hooks/useStore.ts [hunk 2]
```diff
@@ -201,74 +184,37 @@ export const useStore = create<StoreState>()(
...
+          }
         });
       },
-      enteredWorld() {
+      closeWorld() {
         set((state) => {
-          state.world.isEnteredWorld = true;
-          state.world.loadState = WorldLoadState.Entered;
-          state.overlay.isOpen = false;
+          state.world.disposeNetworkInterface?.();
```
[the remark this answers is not in the document]
Comment (ajbura) [reply src/ui/hooks/useStore.ts:208]: creating network interface in mainThread hook with effect is not possible as network interface require groupCall which gets created outside mainThread and moving this inside mainThread hook will complicate error handling. 

So i just moved dispose network  interface call outside zustand store.

**Comment by robertlong** [comment]
Looks great! As you said we need to address the mute state along with the connection reliability when joining while muted. We can do that in another PR

**Review by robertlong — APPROVED** [review]

**Pull request merged**